web/src/components/apps/AppVersionHistory.jsx

```diff
@@ -1330,36 +1386,90 @@ class AppVersionHistory extends Component {
     if (version.preflightResultCreatedAt) {
       newPreflightResults = secondsAgo(version.preflightResultCreatedAt) < 12;
     }
-
     return (
-      <AppVersionHistoryRow
-        key={version.sequence}
-        app={this.props.app}
-        match={this.props.match}
-        history={this.props.history}
-        version={version}
-        selectedDiffReleases={this.state.selectedDiffReleases}
-        nothingToCommit={nothingToCommit}
-        isChecked={isChecked}
-        isNew={isNew}
-        newPreflightResults={newPreflightResults}
-        showReleaseNotes={this.showReleaseNotes}
-        renderDiff={this.renderDiff}
-        toggleShowDetailsModal={this.toggleShowDetailsModal}
-        gitopsEnabled={gitopsEnabled}
-        deployVersion={this.deployVersion}
-        redeployVersion={this.redeployVersion}
-        downloadVersion={this.downloadVersion}
-        upgradeAdminConsole={this.upgradeAdminConsole}
-        handleViewLogs={this.handleViewLogs}
-        handleSelectReleasesToDiff={this.handleSelectReleasesToDiff}
-        renderVersionDownloadStatus={this.renderVersionDownloadStatus}
-        isDownloading={
-          this.state.versionDownloadStatuses?.[version.sequence]
-            ?.downloadingVersion
-        }
-        adminConsoleMetadata={this.props.adminConsoleMetadata}
-      />
+      <React.Fragment key={version.sequence}>
+        <AppVersionHistoryRow
+          handleActionButtonClicked={() =>
```

Comment:
likely that this AppVersionHistoryRow is doing too much for one component if we're passing all these props
let's refactor this later